modules/confiantRtdProvider.js

```diff
@@ -0,0 +1,126 @@
+ * The module will inject a Confiant Inc. script into the page to monitor ad impressions
+ * @module modules/cleanioRtdProvider
+ * @requires module:modules/realTimeData
+ */
+
+import { submodule } from '../src/hook.js';
+import { logError, insertElement, generateUUID } from '../src/utils.js';
+import * as events from '../src/events.js';
+import CONSTANTS from '../src/constants.json';
+
+/**
+ * Injects the Confiant Inc. configuration script into the page, based on proprtyId provided
+ * @param {string} propertyId
+ */
+function injectConfigScript(propertyId) {
+  const scriptElement = document.createElement('script');
+  scriptElement.type = 'text/javascript';
+  scriptElement.src = `https://cdn.confiant-integrations.net/${propertyId}/gpt_and_prebid/config.js`;
+
+  insertElement(scriptElement);
+}
+
+/**
+ * Set up page with Confiant integration
+ * @param {Object} config
+ */
+function setupPage(config) {
+  const propertyId = config?.params?.propertyId;
+  if (!propertyId) {
+    logError('Confiant pbjs module: no propertyId provided');
+    return false;
+  }
+
+  const confiant = window.confiant || Object.create(null);
+  confiant[propertyId] = confiant[propertyId] || Object.create(null);
+  confiant[propertyId].clientSettings = confiant[propertyId].clientSettings || Object.create(null);
+  confiant[propertyId].clientSettings.isMGBL = true;
+  confiant[propertyId].clientSettings.prebidExcludeBidders = config?.params?.prebidExcludeBidders;
+  confiant[propertyId].clientSettings.prebidNameSpace = config?.params?.prebidNameSpace;
+
+  if (window.frames['cnftComm']) {
+    subscribeToConfiantCommFrame(window);
+  } else {
+    setUpMutationObserver();
+  }
+  injectConfigScript(propertyId);
+  return true;
+}
+
+/**
+ * Subscribe to window's message events to report Billable events
+ * @param {Window} targetWindow window instance to subscribe to
+ */
+function subscribeToConfiantCommFrame(targetWindow) {
+  targetWindow.addEventListener('message', reportBillableEvents);
+}
+
+let mutationObserver;
+/**
+ * Set up mutation observer to subscribe to Confiant's communication channel ASAP
+ */
+function setUpMutationObserver() {
+  mutationObserver = new MutationObserver((mutations) => {
+    mutations.forEach((mutation) => {
+      mutation.addedNodes.forEach((addedNode) => {
+        if (addedNode.nodeName === 'IFRAME' && addedNode.name === 'cnftComm' && !addedNode.pbjsModuleSubscribed) {
+          addedNode.pbjsModuleSubscribed = true;
+          mutationObserver.disconnect();
+          mutationObserver = null;
+          const iframeWindow = addedNode.contentWindow;
+          subscribeToConfiantCommFrame(iframeWindow);
+        }
+      });
+    });
+  });
+  mutationObserver.observe(document.head, { childList: true, subtree: true });
+}
+
+/**
+ * Emit billable event when Confiant integration reports that it has monitored an impression
+ */
+function reportBillableEvents (e) {
+  events.emit(CONSTANTS.EVENTS.BILLABLE_EVENT, {
```

Comment:
why bother? doesnt your js already do this emit?

Reply:
@patmmccann no, there we no use cases to do that from our codebase.

Reply:
We're a confiant customer, and you have no trouble billing is, what value is this event adding